### DISCUSSION TITLE
전체적인 이 프로젝트의 문제점들을 적어보기 - 베리

### DISCUSSION THREAD

**Discussion by whk06061** [familymoments/family-moments-android#174] [Ideas]
앞에서 너무 잘 적어주셔서 겹치지 않는 것만 적다보니까 내용이 적네요 ㅎㅎ,,, 🥹

1.  Domain, UI, Data 계층 분리하기 + 계층에 맞는 Model 분리가 필요함
    -> 현재는 Data 계층에서 쓰이는 Model을 UI에서 그대로 쓰고 있는데, 이럴 경우 Data 계층과 UI 계층의 결합이 강해지는 문제가 발생함.
예를 들어 서버에서 내려주는 데이터 구조에 변화가 생기면 UI 계층에서도 수정이 발생하게 됨. 이를 막기 위해 각 계층에 맞는 Model을 두고 Mapping 해줌으로써 결합도를 낮출 수 있음
2. 서버 에러, 클라이언트 에러, 네트워크 에러에 따른 커스텀 에러 타입, 공통 에러 처리 함수 만들기
    -> Data Layer에서 error code에 따라 서버/클라이언트/네트워크 에러 타입을 판단, 리턴해서 하위 레이어에서 적절한 대응이 가능하도록 함
에러 상황에 따른 적절한 에러 메시지를 띄워주거나 UI 대응을 가능하게 하는 것이 좋을 것 같음.
3. 텍스트필드 입력 완료 시 포커스 해제하기

**Comment by boring-km** [comment]
1번을 통해 생기는 문제점이 뭐가 있는지도 같이 적어주면 좋을 것 같아요.
2번을 통해 에러를 구분해서 UI까지 에러 메시지를 보여주는 게 목표가 될까요? 만약 로그로만 남기는 거라면 이 에러로그 수집도 필요할 거 같네요 ㅎㅎ

**Reply by whk06061** [reply]
코비가 말씀하신 부분 내용에 추가해 놓았습니다 👍

**Comment by whk06061** [comment]
저희 프로젝트에서 domain 로직은 Repository 인터페이스가 있는데 Domain Model도 따로 만드는게 저희 프로젝트에 도움이 될지는 확신이 안서서 이 부분에 대해서 얘기 나눠보면 좋을 것 같습니당

**Reply by boring-km** [reply]
Data에 있는 Model을 사용하게 되면, 의존성이 ViewModel에서 바로 Data로 가게 되어서 문제에요.
UiState로 사용하고 있는 객체를 Domain에서 data class로 정의해도 좋을 거 같아요

**Reply by yewon-yw** [reply]
그러면 UI 레이어에서도 UiState를 따로 만들게 되는건가요 아니면 의존성을 갖게 되는건가용?

**Reply by boring-km** [reply]
Domain에서는 Data 모듈에서 API를 통해 필요한 데이터만 뽑은 클래스로 만들고,
UI에서는 로딩이나, 에러 상태를 포함한 UiState가 있으면 되지 않을까 싶네요..!
